vm/ubpf_vm.c

```diff
@@ -685,7 +696,7 @@ ubpf_exec(const struct ubpf_vm* vm, void* mem, size_t mem_len, uint64_t* bpf_ret
             pc += inst.offset;
             break;
         case EBPF_OP_JEQ_IMM:
-            if (reg[inst.dst] == inst.imm) {
+            if (reg[inst.dst] == (uint64_t)sign_extend_immediate(inst.imm)) {
```

Comment:
There is precedent for us having a specialized "conversion" function (e.g., 

```C
static uint32_t
u32(uint64_t x)
{
    return x;
}
```
). Do we want to add something like `u64`? Just a question ... in case it cuts down on code?